libexec/scanner/scanner.c

```diff
@@ -49,12 +49,16 @@
 #include <tsd/sbuf.h>
 #include <tsd/strutil.h>
 #include <tsd/percent.h>
+#include <tsd/timer.h>
 
 struct scan_entry {
 	struct sbuf *path;
 	struct scan_entry *next;
 };
 
+/* record number of stated files for diagnostic purposes */
+static long n_stated_files=0;
+
```

Comment:
I need n_stated_files in tsdfx_scanner() - it is printed there, and it is important information. It is updated in tsdfx_process_dirent() for each stated file. If you do not want a global variable, please suggest which data structure to put it into. Don't see how it can be internal to tsdfx_process_dirent(), which is 2 levels below tsdfx_scanner().

About timer functionality, as I have said earlier - I will most likely add things here, e.g., bytes transferred and bandwidth calculations. I don't want to add too much code all over the place, I want to keep it in one place - e.g., tsd_timer.c. Having it all in one file, and in functions, makes it easier (for you) to remove the functionality if you will not need it, or compile it out all together in production code.

There are many valid programming approaches here, and I do not claim mine is the best. However, it is also not the worst, and I would like to have some space here. If you don't want me to integrate things into your master - I'm happy not to. Maybe start  a branch into which I will be able to commit things, and which you will use to profile file lock. You will be able to remove the stuff you don't want with no effort, or code can be cleaned up later to your liking. Otherwise we risk getting stuck on trivia - note that we are not discussing the core file lock functionality here, but a printf statement.